tests/components/network_sediment_transporter/test_bed_initializer.py

```diff
@@ -0,0 +1,272 @@
+    )
+
+    np.testing.assert_allclose(D50, D50e, rtol=1e-4)
+
+
+class Test_calc_d50_dArea_scaling(object):
+    def test_normal_d50_dArea_scaling(self):
+        """test normal values 1"""
+        D50e = 0.13182567
+        a = 0.1
+        n = 0.12
+        CA = 10  # km^2
+        D50 = calc_d50_dArea_scaling(drainage_area=CA, a=a, n=n)
+        np.testing.assert_allclose(D50, D50e, rtol=1e-4)
+
+    def test_normal_d50_dArea_scaling_2(self):
+        """test normal values 2"""
+        D50e = 0.1
+        a = 0.1
+        n = 0
+        CA = 10  # km^2
+        D50 = calc_d50_dArea_scaling(drainage_area=CA, a=a, n=n)
+        np.testing.assert_allclose(D50, D50e, rtol=1e-4)
+
+
+def test_calc_d50_depth():
+    """test normal values for calc_d50_depth"""
+    D50e = 0.1
+    D50 = calc_d50_depth(
+        slope=0.01,
+        flow_depth=1,
+        tau_c_multiplier=1,
+        rho_water=1000.0,
+        rho_sediment=3000.0,
+        tau_c_50=0.05,
+    )
+    np.testing.assert_allclose(D50, D50e, rtol=1e-4)
+
+
+# %% Test for errors raised
+def test_D50_not_specified(example_nmg2):
+    """test ValueError exception raised if D50 input is None"""
+    with pytest.raises(ValueError):
+        initialize_parcels = BedParcelInitializerUserD50(
+            example_nmg2,
+            user_d50=None,
+        )
+        _ = initialize_parcels()
+
+
+# %% Test for expected correct values
+
+
+class Test_BedParcelInitializer(object):
+    def test_normal_BPI(self, example_nmg2):
+        """
+        Minimum attributes specified, most attributes should use
+        defaults specified at instantiation. uses BedParcelInitializerDischarge.
+        """
+        np.random.seed(seed=5)
+        discharge = np.full(example_nmg2.number_of_links, 1.0)
+        initialize_parcels = BedParcelInitializerDischarge(
+            example_nmg2,
+            discharge_at_link=discharge,
+            median_number_of_starting_parcels=1,
+        )
+
+        parcels = initialize_parcels()
+
+        # create expected values and get values from datarecord
+        GEe = np.expand_dims(["link"] * 8, axis=1).astype(object)
+        GE = parcels.dataset["grid_element"]
+        EIe = np.expand_dims(np.array([0, 1, 2, 3, 4, 5, 5, 6]), axis=1)
+        EI = parcels.dataset["element_id"]
+        SLe = np.array([0, 1, 2, 3, 4, 5, 5, 6])
+        SL = parcels.dataset["starting_link"]
+        ARe = np.ones(8) * 0
+        AR = parcels.dataset["abrasion_rate"]
+        De = np.ones(8) * 2650
+        D = parcels.dataset["density"]
+        TAe = np.expand_dims(np.zeros(8), axis=1)
+        #   TAe = np.array([[ 0.08074127], [ 0.7384403 ], [ 0.44130922], [ 0.15830987],
+        #                 [ 0.87993703], [ 0.27408646], [ 0.41423502], [ 0.29607993]])
+        TA = parcels.dataset["time_arrival_in_link"]
+        ALe = np.expand_dims(np.ones(8), axis=1)
+        AL = parcels.dataset["active_layer"]
+        LL = parcels.dataset["location_in_link"]
+        De = np.array(
```

Comment:
As with a previous comment, it's not clear to me how you know these are the expected values.

Reply:
Hand calculated.